app/services/booking_responder.rb

```diff
@@ -84,4 +86,17 @@ def mark_banned_visitors
       visitor.update! banned: true
     end
   end
+
+  def create_message
+    return nil if booking_response.message_body.blank?
+
+    Message.create!(body: booking_response.message_body,
+                    user: booking_response.user,
+                    visit: visit,
+                    visit_state_change: visit.last_visit_state)
```

Comment:
Yes, I had missed the fact that this was the booking responder class. I would be happier if the state change was passed into the method, but this is definitely less of an issue than I first claimed.